martian/src/mro.rs

```diff
@@ -419,22 +431,96 @@ impl MroDisplay for MroField {
 impl MroField {
     /// Create a new `MroField` with the given name and type.
     /// The field has a default `retain = false`.
-    pub fn new(name: impl ToString, ty: MartianBlanketType) -> Self {
-        fn _new_field(name: String, ty: MartianBlanketType) -> MroField {
+    pub fn new(
+        name: impl ToString,
+        ty: MartianBlanketType,
+        desc: Option<String>,
+        mro_filename: Option<String>,
+    ) -> Self {
+        fn _new_field(
+            name: String,
+            ty: MartianBlanketType,
+            desc: Option<String>,
+            mro_filename: Option<String>,
+        ) -> MroField {
             let field = MroField {
                 name,
                 ty,
+                desc,
+                mro_filename,
                 retain: false,
             };
             field.verify(); // No use case to resultify this so far
             field
         }
-        _new_field(name.to_string(), ty)
+        _new_field(name.to_string(), ty, desc, mro_filename)
     }
+
+    fn name_width(&self) -> usize {
+        if self.mro_filename.is_none() && self.desc.is_none() {
+            self.name.len() + 1 // string length + 1 space
+        } else {
+            self.name.len() // string_length
+        }
+    }
```

Comment:
Not sure it makes sense to be reimplementing this logic rather than just relying on passing the output through `mro format`.  For one thing, this won't actually match the logic used by `mro format`, which is a bit more complex: 
https://github.com/martian-lang/martian/blob/6171ed9de74fad39566b2f08685dcb86d5949eff/martian/syntax/format_callable.go#L116-L121

That is, it does not consider the lengths of names > 35 characters in length, or help texts > 25 characters in length.

Reply:
I suppose we could generate unformatted code and then pass things through MRO format. I know in the past I've had failing GHA tests because the generated MRO code in the repo gets formatted by MRO format and then doesn't match the directly generated code, but I suppose we could just update any such tests to test that it matches generated then MRO formatted code.

Reply:
I agree I had to bend over backwards to get this formatting right and it's much simpler if I don't need to

Reply:
Fair enough! Should we simply auto-generate an mro that has the correct syntax and let the user pipe it through `mro format` downstream?

Reply:
Sounds like both of you agree with that strategy so I will revert it to the unformatted version.
We will need to update rust-martian github action tests to perform mro format when we bump martian-rust in that repo

Reply:
I think it's a good idea to have it at least produce _semi_-formatted output, if for no other reason than to makes it easier to debug if `mro format` chokes on it due to a syntax error.  But I don't think you should go to extreme lengths to replicate the formatting exactly, especially since I reserve the right to change things about the formatting in the future.

Reply:
I think I'll just leave it as-is. It's nice that the output matches MRF closely. and if we have small differences in certain MROs since they aren't exactly the same we can deal with that by applying MRF to case and control in whatever GHA test.